### ISSUE TITLE
Absolutly First Demo (Crack Force Five)

### ISSUE THREAD

**Issue by chris70c** [dirkwhoffmann/vAmiga#612]
Works in WinUAE 512k chip (tested several times, lol) but doesn't in vAmiga, a message "running out of memory" appears before the screen gets corrupted. Can only make it work on vAmiga with ECS 1MB chip.

[Absolutly First Demo (Crack Force Five).zip](https://github.com/dirkwhoffmann/vAmiga/files/7572113/Absolutly.First.Demo.Crack.Force.Five.zip)

**Comment by mras0** [comment]
Try to disable DF1-D3. I think WinUAE behaves the same (be aware that some quickstart options in WinUAE seem to do this automatically). Haven't debugged the EXE but I think it also makes bad assumptions about available chip memory.

Trying it locally I get same behavior with vAmiga only if DF1 is connected. It works without. Same behaviour in WinUAE.

**Comment by chris70c** [comment]
As far as I can tell I only have DF0: connected (at least that's what vAmiga says, lol), I'll try again later :)

**Comment by mras0** [comment]
It works in your currently online version (marked 1.1 Beta 5) at https://www.neoartcr.com/vamiga/ with these settings:

KS1.3 (Rev 34.0005)

![image](https://user-images.githubusercontent.com/9952017/142737450-4b671a68-879d-47dd-a7e3-aea708fc11fd.png)
![image](https://user-images.githubusercontent.com/9952017/142737467-758d9abb-99d3-4f6c-9804-2619e053a805.png)
![image](https://user-images.githubusercontent.com/9952017/142737479-8d62f155-9885-48b0-93e9-cf88316b4132.png)

If I enable DF1 in the Peripherals pane I get the behavior you describe.

This is also the case when I try a local version built from 3798825e ("This is vAmiga 1.1b5") and head 33e24693aed2ffea21943a1044f59aad815dd353 ("Added missing AND mask in setXXXPTL() (#615)"). 

I configure vAmiga like this:
```
        amiga_.revertToFactorySettings();
        amiga_.configure(CONFIG_A500_OCS_1MB);
        amiga_.configure(OPT_SLOW_RAM, /*512*/0); // Intro does not seem to work with this config, same in WinUAE. Just gives blue screen (in wrong color) after flashing LEDs
        amiga_.configure(OPT_BLITTER_ACCURACY, 2);
        //amiga_.configure(OPT_DRIVE_CONNECT, 1, 1); // DF1 - Enabling this gives "running out of memory"
        // load rom here
```

**Comment by chris70c** [comment]
Weird current online version 1.1 Beta 6 :)
Yep same config basically, you can see from the configuration screen that all drives are disabled (and not connected), I guess is another one of those that works randomly one time out of 10 :)

**Comment by mras0** [comment]
Force reloading the page (the correct way this time :)) gave me Beta 6. Still see 100% deterministic behavior: works with DF1 disabled, doesn't work with DF1 enabled.  This is with Firefox 94.0.1 (64-bit) on Windows 11. I tried it like 4-5 times for each config (powering off between each run).

**Comment by chris70c** [comment]
Gonna do some testing with the local version (which is more updated than the online one) and see what happens, I'll let you know asap.

**Comment by chris70c** [comment]
Really weird, I can't make it run with the online or the local version; same config as above, tries with both KS1.2 and 1.3; it always gives the message "running out of memory" then after a while trash the screen and Guru, tried already 20+ times, lol

**Comment by chris70c** [comment]
Found the difference, you have Unmapped Area = All zeroes (which is not the default) try to change it to the default value (which is Floating) and it won't work anymore :)

**Comment by mras0** [comment]
Ahh! If you have the disk inserted on boot it works, but if you wait until the "Insert disk" screen it doesn't!

I see the same behavior in WinUAE (4.9.0b40).

(BTW I also used the "Revert to..." A500 since I could see that I had "Init pattern" set to random, though this didn't make a difference).

**Comment by chris70c** [comment]
I guess in any case is not really the emulator fault so we can probably close this issue (might worth maybe investigate if it is better to set Unmapped Area = All zeroes by default like we did for Init pattern).

**Comment by mras0** [comment]
I think if you can reproduce my finding, i.e. it works if disk is inserted before power on, but not if inserted later, then yes, I think you can close it since it matches WinUAE behavor (and my emulator FWIW, which isn't much). Also seems plausible given what I've seen from a quick disassembly of the code.

However for me the unmapped  area part didn't make a difference, so I'd be hesitant to draw any conclusions there. I've also tried to locally reproduce some of the earlier differences you've reported regarding the RAM init pattern, and I can't reproduce them. Maybe there were some improvements in vAmiga that fixed them.

Maybe it would be a good idea to re-check all of the issues that seemed to have been cause by those settings and list the ones that are still current in https://github.com/dirkwhoffmann/vAmiga/issues/586 ? I think if there are really emulator differences depending on RAM init pattern/unmapped memory it's likely due to a vAmiga bug or exciting emulation difference that might be worth looking into (or buggy intro/demo of couse).

**Comment by chris70c** [comment]
Unmapped Area = Floating, Disk inserted after KS hand = doesn't work
Unmapped Area = Floating, Disk alreay inserted = works
Unmapped Area = All zeroes, Disk inserted after KS hand = doesn't work
Unmapped Area = All zeroes, Disk already inserted = works

so yes the differences is definitely if the disk is already inserted or not,
probably when I changed the config I left the disk in the drive and just
resetted.

**Comment by chris70c** [comment]
Yes I can confirm that all the intros that before required Init pattern = All zeroes are now working with any setting just fine, they no longer recognize a NTSC machine as before even with Init pattern = Random.
I guess something has changed in the latest code and that the Init pattern was not the root cause of the problem.

**Comment by dirkwhoffmann** [comment]
Fixed in v1.1b8
